nova/tests/unit/virt/vmwareapi/test_volumeops.py

```diff
@@ -57,8 +57,10 @@ def setUp(self):
             **self._instance_values)
 
     def _test_detach_disk_from_vm(self, destroy_disk=False):
-        def fake_call_method(module, method, *args, **kwargs):
-            vmdk_detach_config_spec = kwargs.get('spec')
+        def fake_call_method(module, method, *args, spec=None, **kwargs):
+            if spec is None:
+                raise ValueError("Needs spec")
```

Comment:
```
>>> def a(b, *args, c, **kwargs):
...     print(b, args, c, kwargs)
... 
>>> a()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: a() missing 1 required positional argument: 'b'
>>> a(1)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: a() missing 1 required keyword-only argument: 'c'
>>> a(1, 2)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: a() missing 1 required keyword-only argument: 'c'
>>> a(1, 2, c=3)
1 (2,) 3 {}
```
No need for the check, just get rid of the `=None`.

Reply:
Changed the code accordingly